### PR TITLE
docs isr_rules, micropython: add note that creating a ref to a bound method causes allocation.

### PR DESCRIPTION
**Pull request by peterhinch**
As discussed in #3428.

### CODE REVIEW

**Comment by pfalcon** [comment]
This has stray commits, please rebase.

File: docs/library/micropython.rst [hunk 1]
```diff
@@ -111,6 +111,13 @@ Functions
    Such an IRQ puts restrictions on the code that runs in the IRQ (for example
    the heap may be locked) and scheduling a function to call later will lift
    those restrictions.
+   
+   If the callback to be passed to ``schedule`` is a bound method, passing this
+   directly will fail. This is because creating a reference to a bound method
```
Comment (pfalcon) [inline docs/library/micropython.rst]: "a bound method, passing this directly will fail." This isn't true in general. The paragraph apparently should start with "Note: If schedule is call from a preempting IRQ, when memory allocation is not allowed, ..."

**Comment by peterhinch** [comment]
Has this fixed it? Git continues to baffle me, I'm afraid :(

**Comment by pfalcon** [comment]
Nope, as you can see, there're still 2 commits.

In your "doc-ref-bound-method" branch, you need to do "git rebase -i master", then follow the on-screen instructions. (You need to remove a commit not related to this patch.)

**Comment by peterhinch** [comment]
I had to do a force push but hopefully this has worked.

File: docs/reference/isr_rules.rst [hunk 2]
```diff
@@ -124,6 +124,30 @@ A means of creating an object without employing a class or globals is as follows
 The compiler instantiates the default ``buf`` argument when the function is
 loaded for the first time (usually when the module it's in is imported).
 
+An instance of object creation occurs when a reference to a bound method is created. This means that
```
Comment (pfalcon) [inline docs/reference/isr_rules.rst]: Can you please wrap lines similarly to the paragraph above. (I also would recommend wrapping them at <80 chars for any new paragraphs).

File: docs/library/micropython.rst [hunk 3]
```diff
@@ -112,5 +112,14 @@ Functions
    the heap may be locked) and scheduling a function to call later will lift
    those restrictions.
 
+   Note: If ``schedule`` is called from a preempting IRQ, when memory
```
Comment (pfalcon) [inline docs/library/micropython.rst]: Can you please use the xref syntax as described in https://github.com/micropython/micropython/blob/master/CODECONVENTIONS.md#documentation-conventions (specifically, all occurrences should be:
~~~
`schedule()`
~~~

**Comment by peterhinch** [comment]
Is this OK now?

**Comment by dpgeorge** [comment]
Thanks!  Merged in ec1e9a10a73d9c5c3771a5797e9eceb62475b197

**Pull request closed without merging**